tests/src/unit-bjdata.cpp

```diff
@@ -2815,6 +2815,126 @@ TEST_CASE("BJData")
+                    {
+                        return true;
+                    }
+                    bool number_float(json::number_float_t, const std::string&) override
+                    {
+                        return true;
+                    }
+                    bool string(std::string&) override
+                    {
+                        return true;
+                    }
+                    bool binary(json::binary_t&) override
+                    {
+                        return true;
+                    }
+                    bool start_object(std::size_t) override
+                    {
+                        return true;
+                    }
+                    bool key(std::string&) override
+                    {
+                        return true;
+                    }
+                    bool end_object() override
+                    {
+                        return true;
+                    }
+                    bool start_array(std::size_t) override
+                    {
+                        return true;
+                    }
+                    bool end_array() override
+                    {
+                        return true;
+                    }
+
+                    // This is the only method we care about - detecting error 408
+                    bool parse_error(std::size_t, const std::string&, const json::exception& ex) override
+                    {
+                        if (ex.id == 408)
+                        {
+                            overflow_detected = true;
+                        }
+                        return false;
+                    }
+                };
+
+                // Create BJData payload with overflow-causing dimensions (2^32+1) × (2^32)
+                std::vector<uint8_t> bjdata_payload =
+                {
+                    0x5B,                                           // '[' start array
+                    0x24, 0x55,                                     // '$', 'U' (type uint8)
+                    0x23, 0x5B,                                     // '#', '[' (dimensions array)
+                    0x4D,                                           // 'M' (uint64)
+                    0x01, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, // 2^32 + 1 (4294967297) as little-endian
+                    0x4D,                                           // 'M' (uint64)
+                    0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, // 2^32 (4294967296) as little-endian
+                    0x5D                                            // ']' end dimensions
+                    // No data - we don't need it for this test, we just want to hit the overflow check
+                };
+
+                // Test with overflow dimensions using SAX parser
+                {
+                    SimpleOverflowSaxHandler handler;
+                    auto result = json::sax_parse(bjdata_payload, &handler,
```

Comment:
This variable is not used which is a warning/error in the MSVC build.

Reply:
Yup, will push a fix soon!

Reply:
Thanks! There are also some warnings/errors unrelated to your changes which will be fixed with #4763.